src/pages/CubePlaytestPage.js

```diff
@@ -272,38 +273,26 @@ const CubePlaytestPage = ({ cube, cubeID, canEdit, decks, draftFormats }) => {
 
   const handleDeleteFormat = useCallback(
     async (event) => {
-      const formatIndex = parseInt(event.target.getAttribute('data-index'), 10);
+      const formatID = event.target.getAttribute('data-id');
       try {
-        const response = await csrfFetch(`/cube/format/remove/${cubeID};${formatIndex}`, {
+        const response = await csrfFetch(`/cube/${cubeID}/format/${formatID}`, {
           method: 'DELETE',
         });
-        if (!response.ok) throw Error();
+        if (!response.ok) throw Error(response.statusText);
 
         const json = await response.json();
         if (json.success !== 'true') throw Error();
 
-        addAlert({
-          color: 'success',
-          message: 'Format successfully deleted.',
-        });
-        setFormats(formats.filter(({ index }) => index !== formatIndex));
+        addAlert('success', 'Format successfully deleted.');
+        setFormats(formats.filter((format) => format.id !== formatID));
       } catch (err) {
         console.error(err);
-        addAlert({
-          color: 'danger',
-          message: 'Failed to delete format.',
-        });
+        addAlert('danger', `Failed to delete format. ${err.message}`);
       }
     },
     [addAlert, cubeID, formats],
   );
 
-  // Sort formats alphabetically.
```

Comment:
I’d still prefer to do all sorting on the client, even with robust IDs. Layout should be done client side.

Reply:
Up to you, I agree that sorting should be possible on the client, but generally you should be able to request an order from a server or have it defaulted to a particular deterministic order. If the client changes that order everything should still work (hence the use of IDs and never index).